pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,719 @@
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/nspcc-dev/neo-go/pkg/vm/vmstate"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/nspcc-dev/neofs-contract/nns"
+	"go.uber.org/zap"
+)
+
+// Blockchain provides Neo blockchain services consumed by NeoFS Inner Ring. By
+// design, Blockchain does not implement Inner Ring specifics: instead, it
+// provides the generic functionality of the Neo blockchain, and narrows the
+// rich Neo functionality to the minimum necessary for the operation of the
+// Inner Ring.
+//
+// Blockchain must be initialized using New constructor. After initialization
+// Blockchain becomes a single-use component that can be started and then
+// stopped. All operations should be executed after Blockchain is started and
+// before it is stopped (reverse behavior is undefined).
+type Blockchain struct {
+	base *core.Blockchain
+
+	netServer *network.Server
+
+	nep17GAS *nep17.Token
+
+	actor *actor.Actor
+
+	rpcActor notary.RPCActor
+
+	chErr chan error
+}
+
+// SignTools groups signing components for New constructor.
+type SignTools struct {
+	// NEO account of the node.
+	Account *wallet.Account
+
+	// File containing Account.
+	AccountFile struct {
+		// Path to the file.
+		Path string
+		// Account password.
+		Password string
+	}
+}
+
+// New returns new Blockchain instance initialized using provided parameters. If
+// New returns an error, Blockchain can (and must) not be used with provided
+// configuration.
+//
+// Provided file path is expected to lead to the Neo Go partial (*)
+// configuration file: it is used to build core blockchain components similar to
+// the Neo Go node in order to use Neo blockchain services.
+// (*) partial means that some values are static and can't be configured:
+//   - ApplicationConfiguration.Relay is always set
+//   - ApplicationConfiguration.X.UnlockWallet is set to specified SignTools.AccountFile
+//     (X - Consensus, P2PNotary)
+//   - ApplicationConfiguration.X.Enabled is set if node belongs to
+//     the committee (X - Consensus, P2PNotary)
+//
+// Given SignTools.Account is expected to be Inner Ring node's NEO account. It
+// must be previously unlocked for signing, and SignTools.AccountFile must
+// correspond to it. If the account belongs to the committee of the configured
+// Neo blockchain, resulting Blockchain is run in consensus mode.
+//
+// Given zap.Logger instance is used to log Blockchain events. It should be
+// previously initialized: Blockchain does not modify the logger.
+//
+// Specified error channel is used to report internal Blockchain errors to the
+// superior context (Inner Ring application) listening to it in the background.
+func New(neoNodeConfigFilepath string, signTools SignTools, log *zap.Logger, chErr chan<- error) (*Blockchain, error) {
```

Comment:
Initially I decided to completely merge IR and Neo Go configs. But since there is no need to configure some neo-go parameters in IR, this parameter can be made a local structure.